third_party/mesa3d/src/util/rust/sys/linux/descriptor.rs

```diff
@@ -54,6 +60,21 @@ impl OwnedDescriptor {
             }
         }
     }
+
+    fn get_memory_handle_type(&self) -> Result<u32> {
+        let fd_path = read_link(format!("/proc/self/fd/{}", self.as_raw_descriptor()))
+            .map_err(|_| Error::from(ErrorKind::Unsupported))?;
+
+        let path_str = fd_path.to_string_lossy();
+        if path_str.starts_with("/dmabuf:") {
+            Ok(MESA_HANDLE_TYPE_MEM_DMABUF)
+        } else if path_str.starts_with("/memfd:") {
+            Ok(MESA_HANDLE_TYPE_MEM_SHM)
+        } else {
+            // Default to SHM for unknown types
+            Ok(MESA_HANDLE_TYPE_MEM_SHM)
+        }
+    }
```

Comment:
nice!, can you send this to 

- https://gitlab.freedesktop.org/mesa/mesa/:
- https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/util/rust/sys/linux/descriptor.rs


mesa3d is the canonical upstream for this code.

Reply:
Sure, Opened it at https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/38677 
Should I wait for mesa3d to merge before this PR, or can we merge this PR first and sync from mesa3d later?